examples/cryptocurrency-advanced/backend/src/schema.rs

```diff
@@ -14,94 +14,106 @@
 
 //! Cryptocurrency database schema.
 
-use exonum_merkledb::{IndexAccess, ObjectHash, ProofListIndex, ProofMapIndex};
+use exonum_merkledb::{AccessExt, IndexAccessMut, ObjectHash, ProofListIndex, ProofMapIndex};
 
 use exonum::crypto::{Hash, PublicKey};
 
 use crate::{wallet::Wallet, INITIAL_BALANCE};
 
+const NOT_INITIALIZED: &str = "Cryptocurrency schema is not initialized";
+
 /// Database schema for the cryptocurrency.
 #[derive(Debug)]
-pub struct Schema<'a, T> {
-    service_name: &'a str,
+pub struct Schema<T: AccessExt> {
```

Comment:
I called the trait `AccessExt` because it initially was purely an extension trait for `IndexAccess`. This is no longer the case (`Prefixed<_>` implements `AccessExt`, but not `IndexAccess`; I've tried this way btw, but it turned ugly really fast), so the name may indeed be confusing. `Access` could probably be a good name (I'd rename `IndexAccess` to something like `RawAccess` or `ViewAccess` as well).